lib/ts/recipe/session/with-jwt/recipeImplementation.ts

```diff
@@ -42,6 +43,61 @@ export default function (
         return accessTokenExpiry + EXPIRY_OFFSET_SECONDS;
     }
 
+    async function jwtAwareUpdateAccessTokenPayload(
+        sessionInformation: SessionInformation,
+        newAccessTokenPayload: any,
+        userContext: any
+    ) {
+        let accessTokenPayload = sessionInformation.accessTokenPayload;
+
+        let existingJwtPropertyName = accessTokenPayload[ACCESS_TOKEN_PAYLOAD_JWT_PROPERTY_NAME_KEY];
+
+        if (existingJwtPropertyName === undefined) {
+            return await originalImplementation.updateAccessTokenPayload({
+                sessionHandle: sessionInformation.sessionHandle,
+                newAccessTokenPayload,
+                userContext,
+            });
+        }
+
+        let existingJwt = accessTokenPayload[existingJwtPropertyName];
+        assert.notStrictEqual(existingJwt, undefined);
+
+        let currentTimeInSeconds = Date.now() / 1000;
+        let decodedPayload = JsonWebToken.decode(existingJwt, { json: true });
+
+        // JsonWebToken.decode possibly returns null
+        if (decodedPayload === null) {
+            throw new Error("Error reading JWT from session");
+        }
+
+        let jwtExpiry = decodedPayload.exp - currentTimeInSeconds;
+
+        if (jwtExpiry <= 0) {
+            // it can come here if someone calls this function well after
+            // the access token and the jwt payload have expired. In this case,
+            // we still want the jwt payload to update, but the resulting JWT should
+            // not be alive for too long (since it's expired already). So we set it to
+            // 1 second lifetime.
+            jwtExpiry = 1;
+        }
+
+        newAccessTokenPayload = await addJWTToAccessTokenPayload({
+            accessTokenPayload: newAccessTokenPayload,
+            jwtExpiry,
+            userId: sessionInformation.userId,
+            jwtPropertyName: existingJwtPropertyName,
+            openIdRecipeImplementation,
+            userContext,
+        });
+
+        return await originalImplementation.updateAccessTokenPayload({
```

Comment:
Don't call to `originalImplementation` require bind(this)?

Reply:
Huh? I don't think i understood why not.

Reply:
It'd only make a difference if the original implementation calls anything we provide an override for here, which is not the case + we don't add this for other methods (see line 102). 
Besides, this is handled by the override builder it handles all binding.